bgpd/bgp_nexthop.c

```diff
@@ -48,43 +47,66 @@
 
 DEFINE_MTYPE_STATIC(BGPD, MARTIAN_STRING, "BGP Martian Address Intf String");
 
-char *bnc_str(struct bgp_nexthop_cache *bnc, char *buf, int size)
+static inline int bgp_nexthop_cache_compare(const struct bgp_nexthop_cache *a,
+					    const struct bgp_nexthop_cache *b)
 {
-	prefix2str(bgp_dest_get_prefix(bnc->dest), buf, size);
-	return buf;
+	return prefix_cmp(&a->prefix, &b->prefix);
+}
+
+RB_GENERATE(bgp_nexthop_cache_head, bgp_nexthop_cache, entry,
```

Comment:
Shouldn't this be DECLARE_RBTREE from typesafe.h?  I would prefer we standardize on the new data structures instead of introducing code that uses the old standards

Reply:
Yes, this should be a `DECLARE_RBTREE_UNIQ`. Unfortunately we used the old data structures everywhere (also in pathd) bc we were not aware of the newer ones :(.

@sylane this is what we are talking about here: http://docs.frrouting.org/projects/dev-guide/en/latest/lists.html

I think we can change it here somewhat easily.